docs/orleans/grains/grain-extensions.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Grain extensions
+description: Learn how to extend an Orleans Grain.
+ms.date: 05/08/2023
+---
+
+# Grain extensions
+
+Grain extensions provide a way to add extra behavior to grains. By extending a grain with an interface that derives from <xref:Orleans.Runtime.IGrainExtension>, you can add new methods and functionality to the grain.
+
+In this article, you see two examples of grain extensions. The first example shows how to add a `Deactivate` method to all grains that can be used to deactivate the grain. The second example shows how to add a `GetState` and `SetState` method to any grain, allowing you to manipulate the grain's internal state.
+
+## Deactivate extension example
+
+In this example, you learn how to add a `Deactivate` method to all grains automatically. The method can be used to deactivate the grain and accepts a string as a message parameter.
```

Comment:
I wonder if it's worth noting that grains support this out of the box via `IGrainManagementExtension`. The example is still a worthwhile one, but users should never need to build this particular one in practice.